### PR TITLE
3213 Selected tab underline doesn't move correctly with SidePanel open/close

### LINKED ISSUE #3213: Selected tab underline doesn't move correctly with SidePanel open/close
## What went wrong? 😲
 

Not hindering anything, just a funky lil UX moment.

Details tab is underlined: 
Screenshot 2024-03-07 at 11 13 55 AM

I open the side panel... the underline didn't follow the `Details` tab name 😢:
Screenshot 2024-03-07 at 11 14 01 AM

Close the side panel... now it moves! So it's in the wrong place AGAIN!
Screenshot 2024-03-07 at 11 14 05 AM

Without looking into it, I am slightly surprised that the tab titles move, but their underline doesn't lol. 

## Expected behaviour 🤔

Active tab remains underlined as the side panel is opened and closed.

## How to Reproduce 🔨

Steps to reproduce the behaviour:

On a detail view (e.g. an inbound shipment), open and close the `More` side panel. See the tab underline 

## Your environment 🌱

- Version: 1.7.0

- Platform: browser (chrome)

### PR DESCRIPTION
**Pull request by roxy-dao**
Fixes #3213

# 👩🏻‍💻 What does this PR do? 
Fixes underline being in funky places

# 🧪 How has/should this change been tested? 
- [ ] Go to invoice
- [ ] Open side panel by clicking more
- [ ] Check that the underline is still in the correct place 
- [ ] Can even move tabs

### CODE REVIEW

**Comment by clemens-msupply** [comment]
You just removed the debounce, right? but why is this working? or why wasn't it working before?

**Comment by roxy-dao** [comment]
> You just removed the debounce, right? but why is this working? or why wasn't it working before?

debounce and useCallback aren't working well together. We either use one or the other 👀  originally it was just debounced. useCallback got added sometime with no explanation...

File: client/packages/common/src/ui/components/navigation/Tabs/DetailTabs.tsx [hunk 1]
```diff
@@ -42,17 +41,19 @@ export const DetailTabs: FC<DetailTabsProps> = ({
...
-  const handleResize = useCallback(
-    () => debounce(() => window.dispatchEvent(new Event('resize')), 100),
-    []
-  );
+  const handleResize = useCallback(() => {
+    window.dispatchEvent(new Event('resize'));
+  }, []);
+  useEffect(() => {
+    handleResize();
+  }, [detailPanelOpen, drawerOpen, handleResize]);
```
Comment (clemens-msupply) [inline client/packages/common/src/ui/components/navigation/Tabs/DetailTabs.tsx:56]: This doesn't work for me, the bar is not centerred correctly...

```ts
useEffect(() => {
    setTimeout(() => {
      window.dispatchEvent(new Event('resize'));
    }, 100);
  }, [detailPanelOpen, drawerOpen]);
```
This works better though. Don't ask me why...

@mark-prins where is that underline drawn? is that in MUI or are we doing it?

Comment (roxy-dao) [reply client/packages/common/src/ui/components/navigation/Tabs/DetailTabs.tsx:56]: But work the same for me... I see no difference in the underline movement

Comment (clemens-msupply) [reply client/packages/common/src/ui/components/navigation/Tabs/DetailTabs.tsx:56]: Without the debounce or the delay the event seem to be created to early, i.e. while the sidebar is still expanding/shrinking. Thus the underline stops to early...

Comment (clemens-msupply) [reply client/packages/common/src/ui/components/navigation/Tabs/DetailTabs.tsx:56]: Just a guess: there are multiple debounce calls and the time of the first debounce call + the delay is still before the animation finished, thus it doesn't work for me. My machine is slower than yours, so I can see this and you can't (?).

File: client/packages/common/src/ui/components/navigation/Tabs/DetailTabs.tsx [hunk 2]
```diff
@@ -42,17 +41,21 @@ export const DetailTabs: FC<DetailTabsProps> = ({
...
   const { isOpen: detailPanelOpen } = useDetailPanelStore();
   const { isOpen: drawerOpen } = useDrawer();
-  const { showConfirmation } = useConfirmOnLeaving(false);
-  const handleResize = useCallback(
-    () => debounce(() => window.dispatchEvent(new Event('resize')), 100),
-    []
-  );
+  // const handleResize = useCallback(() => {
+  //   window.dispatchEvent(new Event('resize'));
+  // }, []);
```
Comment (clemens-msupply) [inline client/packages/common/src/ui/components/navigation/Tabs/DetailTabs.tsx:53]: Some dead code....

**Review by clemens-msupply — APPROVED** [review]
Looks good, just remove the dead code please

**Pull request merged**